hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AbstractAmqpServiceIntegrationTest.java

```diff
@@ -196,15 +215,67 @@ private void assertAssignmentMessage(final Set<SoftwareModule> dsModules, final
         assertThat(updatedTarget.getSecurityToken()).isEqualTo(downloadAndUpdateRequest.getTargetSecurityToken());
     }
 
-    protected void assertDownloadAndInstallMessage(final Set<SoftwareModule> dsModules, final String controllerId) {
-        assertAssignmentMessage(dsModules, controllerId, EventTopic.DOWNLOAD_AND_INSTALL);
+    protected void assertDownloadAndInstallMessage(final Set<SoftwareModule> softwareModules,
+            final String controllerId) {
+        assertAssignmentMessage(softwareModules, controllerId, EventTopic.DOWNLOAD_AND_INSTALL);
+
+    }
+
+    protected void assertLatestMultiActionMessageContainsInstallMessages(final String controllerId,
+            final List<Set<Long>> smIdsOfActionsExpected) {
+        final Message multiactionMessage = replyToListener.getLatestEventMessage(EventTopic.MULTI_ACTION);
+        assertThat(multiactionMessage.getMessageProperties().getHeaders().get(MessageHeaderKey.THING_ID))
+                .isEqualTo(controllerId);
+        final DmfMultiActionRequest multiActionRequest = (DmfMultiActionRequest) getDmfClient().getMessageConverter()
+                .fromMessage(multiactionMessage);
+
+        final List<Set<Long>> smIdsOfActionsFound = getDownloadAndUpdateRequests(multiActionRequest).stream()
+                .map(AbstractAmqpServiceIntegrationTest::getSmIds).collect(Collectors.toList());
+        assertThat(smIdsOfActionsFound).containsExactlyElementsOf(smIdsOfActionsExpected);
```

Comment:
I thin order doesn't matter here, right?
```suggestion
        assertThat(smIdsOfActionsFound).containsExactlyInAnyOrderElementsOf(smIdsOfActionsExpected);
```